### PR TITLE
feat: Navigation Tabs

### PR DESCRIPTION
**Pull request by vivek-p44**
Closes # https://project44.atlassian.net/browse/MI-2

## 📝 Description

> Added **NavigationTabs** component in manifest.

## Screenshots

> Please provide screenshots for any visual changes

## Merge checklist

- [x] Added/updated tests
- [x] Added changeset

### CODE REVIEW

File: packages/react/src/components/NavigationTabs/CountLabel/index.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  css?: CSS;
+  isLoading?: boolean;
+}
+
+export function CountLabel({ value, variant = 'default', dense = false, css, isLoading }: Props) {
+  const theme = () => {
+    switch (variant) {
+      case 'black':
+        return {
+          backgroundColor: designTokens.colorTextPrimary,
```
Comment (vitaliy-p44) [inline packages/react/src/components/NavigationTabs/CountLabel/index.tsx:20]: can't we use string tokens here? like "$colors$text-primary"

File: packages/react/src/components/NavigationTabs/index.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    | 'default'
+    | 'grey'
+    | 'lightGrey'
+    | 'red';
+}
+
+const activeTabStyle: CSS = {
+  background: designTokens.colorPaletteWhite,
+  color: designTokens.colorBackgroundTopNav,
+};
```
Comment (vitaliy-p44) [inline packages/react/src/components/NavigationTabs/index.tsx:48]: same

File: packages/react/src/components/NavigationTabs/CountLabel/index.tsx (hunk 1 above)
Comment (vivek-p44) [reply packages/react/src/components/NavigationTabs/CountLabel/index.tsx]: Thanks @vitaliy-p44 for pointing this, Updated code accordingly.

File: packages/react/src/components/NavigationTabs/index.tsx (hunk 2 above)
Comment (vivek-p44) [reply packages/react/src/components/NavigationTabs/index.tsx:47]: Done ✅

**Comment by vitaliy-p44** [comment]
It looks kind of weird If don't pass any additional styles props.
`<NavigationTabs tabs={tabData} />`

image

Maybe we should have better style by default?
@dmorton-p44 @p44dyoachim thoughts?

**Comment by vivek-p44** [comment]
@vitaliy-p44 @dmorton-p44 @p44dyoachim Any update on this ?
CC: @smartinelli-p44 @aamir-p44

File: packages/react/src/components/NavigationTabs/CountLabel/index.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        return {
+          backgroundColor: '$palette-red-100',
+          color: '$palette-red-900',
+        };
+      default:
+        return {
+          backgroundColor: '$palette-black',
+          color: '$palette-white',
+        };
+    }
```
[the remark this answers is not in the document]
Comment (vivek-p44) [reply packages/react/src/components/NavigationTabs/CountLabel/index.tsx]: Thanks for suggestion @romulof, Done ✅

File: packages/react/src/components/NavigationTabs/index.tsx [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+  } = props;
+  const [activeTab, setActiveTab] = useState(defaultActive);
+
+  useEffect(() => void setActiveTab(defaultActive), [defaultActive]);
+
+  return (
+    <Flex
+      ref={forwardedRef}
+      align="center"
+      aria-label="Navigation Tabs"
```
[the remark this answers is not in the document]
Comment (vivek-p44) [reply packages/react/src/components/NavigationTabs/index.tsx]: Done ✅

File: packages/react/src/components/NavigationTabs/index.tsx [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        const isActive = activeTab === index;
+        return (
+          <Tooltip
+            key={toolTip}
+            css={{
+              marginLeft: '80px',
+              width: '350px',
+            }}
+            data-testid="navigation-tooltip"
+            delay={250}
```
[the remark this answers is not in the document]
Comment (vivek-p44) [reply packages/react/src/components/NavigationTabs/index.tsx:82]: @romulof Took reference from existing implementation of navgiation tabs in mwa.

**Comment by vivek-p44** [comment]
> It looks kind of weird If don't pass any additional styles props. `<NavigationTabs tabs={tabData} />`
> 
> image
> Maybe we should have better style by default? @dmorton-p44 @p44dyoachim thoughts?

Updated default style to general use case we have, for particular use case it can be overwritten with css.
Screenshot 2023-12-06 at 10 04 00 AM

File: apps/website/docs/navigation/navigation-tabs.mdx [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+  return <NavigationTabs tabs={tabData} css={style.tabsStyle} />;
+}
+```
+
+## With Counter Button Tabs
+
+We can show a count of result for a particular button by passing count property.
+
+```jsx live
+function NavigationTabsDemo() {
```
Comment (vitaliy-p44) [inline apps/website/docs/navigation/navigation-tabs.mdx:44]: Can we remove tons of css from the example too?

**Review by vitaliy-p44 — APPROVED** [review]

**Pull request merged**